### ISSUE TITLE
Usage in different area

### ISSUE THREAD

**Issue by ehvid** [boathit/deepgtt#5]
Hello Xiucheng!
We have been trying to implement DeepGTT on the DiDi dataset of Chengdu, but have encountered an error when running the train.py file; `mat1 dim 1 must match mat2 dim 0`.

Would you happen to know, what may be causing this error, and how we can proceed?

To adapt to Chengdu instead of Harbin, we have have followed the readme, though with the following differences;
- Mapmatched using the Barefoot mapmatched with the changes to the import.bash file to cover Chengdu. We converted the DiDi files from .txt to .h5 for usage (in the process converted the GPS coordinates from GCJ to WGS).
- Run the gentrain.py with the hyper-parameters.JSON file modified to fit Chengdu. 

Apologies for the rather vague request, we have not changed much besides these changes. We are in a bit of stump, and have a hard time figuring out how to proceed from this error. 

Thank you very much in advance,

Regards Emil

**Comment by boathit** [comment]
Not sure if it is caused by PyTorch version difference. Can you run it normally with the Harbin dataset?

**Comment by ehvid** [comment]
Thank you for your hastly reply, very appreciated. 
Unfortunally, I have tried running the Harbin dataset, and it seems to work correctly, so no fault there.
On further inspection of my dataset, and the .h5 files resulting from the gentrain.jl, the 'S' (the tensorcore?) seems to have negative values, this cannot be correct, can it?

**Comment by boathit** [comment]
It should not be negative. This [line](https://github.com/boathit/deepgtt/blob/master/harbin/julia/traindata.jl#L54) is hard coded, perhaps you need to change the year value and also rename you file in the format "yy_mm_dd.h5" e.g. "15_10_03.h5".

**Comment by ehvid** [comment]
Think you for your reply. I have tried cleaning up my dataset, sorting the dataset according to timestamps, which solved the negative values. Furthermore, I have changed such that the names are foramtted as you mentioned, and are parsed correctly as month and day. 

Unfortunaly, I still recieve the same error messages as previously. I was looking into the differences of the Harbin and Chengdu (DiDi) data sets, and it seems the trips of Chengdu are vastly longer, i.e. more gps-points; this should not make a difference though, should it?

**Comment by ehvid** [comment]
Also, should the tensor-core have specific dimensions?

**Comment by boathit** [comment]
It should not be a problem with high sampling rates because the preprocessing script will remove the redundant points, [`removereduantpoints`](https://github.com/boathit/deepgtt/blob/master/harbin/julia/Trip.jl#L136). 

Maybe you can compare the small batch of data returned by the [`DataLoader`](https://github.com/boathit/deepgtt/blob/master/harbin/python/train.py#L150) of the two cities.

**Comment by ehvid** [comment]
I ended up hardcoding the size of the [regions I, O S and C](https://github.com/boathit/deepgtt/blob/16fc03f47a02c41b2b2ef0500145c3fb62336ee9/harbin/julia/SpatialRegion.jl#L49) to fit the size of the one created in the Harbin-set (`zeros(Float32, 138, 148)`).
I will admit, I am not sure if this will affect the final result of the trained model?

**Comment by ehvid** [comment]
Also, when adding the validation and test data, how should the various files be distributed? Is a single file enough for respectively test and validation, or do you recomend a ration between training, test and validation files?
The DiDi set in question is 30 days, i.e. 30 files.

**Comment by learnurban** [comment]
@ehvid Excuse me, do you have some results for deepGTT in the Chengdu DiDi dataset? I can share my code, and exchange experiences with you.

**Comment by ehvid** [comment]
@NKzhangzhiwen, sorry, unfortunately not. Out project has since ended with the end of the semester. Hope your problems are resolved.
